presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java

```diff
@@ -35,7 +35,10 @@
     @Deprecated
     default PlanNode resolve(PlanNode node)
     {
-        return getOnlyElement(resolveGroup(node));
+        if (node instanceof GroupReference) {
```

Comment:
I'm unsure about this change. It makes `resolve` (old) & `resolveGroup` (new) _potentially_ inconsistent. OTOH,  according to `resolveGroup`'s javadoc, this should not change anything.

Reply:
I am sorry I did not read the java doc, I have just read the method signature. `resolveGroup` suggests it resolves group, so IMO it should only accept `GroupReference`. Hence I think that javadoc should be fixed and `resolveGroup` should have a checkArgument added (or signature changed).

Calling this for regular node is pointless and it could hide some bugs in rules as it is not a defensive (strict) approach.

What do you think?

Reply:
Anyway,  I am going to remove this change  from this PR. It is more or less a design change, not a small refactor.